controllers/certificatesigningrequests_controller.go

```diff
@@ -102,8 +105,34 @@ func (r *certificateSigningRequestsReconciler) Reconcile(ctx context.Context,
 	return ctrl.Result{}, r.reconcileCSR(certificateSigningRequest)
 }
 
-// reconcileCSR handles the CSR validation and approval
+// reconcileCSR kicks off CSR validation and approval. In case of update conflicts, the approval process is wrapped
+// with retry logic using the Update --> Get --> Retry Update pattern. Retry will occur until specified timeout
 func (r *certificateSigningRequestsReconciler) reconcileCSR(request *certificates.CertificateSigningRequest) error {
+	err := wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
+		err := r.validateAndApprove(request)
+		if err != nil {
+			if k8sapierrors.IsConflict(err) {
+				r.log.Info("Update conflict when attempting to approve CSR, retrying...", "name", request.Name,
```

Comment:
Sorry I am not sure the link is resolving properly, it's not clear to me how the name and namespace would already be present.

Reply:
I was trying to link to this line:
```
	_ = r.log.WithValues(CSRController, req.NamespacedName)
```

Reply:
Conclusion here is that the name and namespace will not be propagated to the log output because we are using the `logr.WithValues` method improperly. [Tech debt story](https://issues.redhat.com/browse/WINC-721) created as that is out of PR scope